src/pages/mypage/profile.tsx

```diff
@@ -1,38 +1,109 @@
-import { useState } from 'react';
+import { useMemo, useState } from 'react';
+import { USER_QUERY_KEY, useUserQuery } from '../../api/hook/users';
 import Button from '../../components/common/button';
 import TextInput from '../../components/common/form/text-input';
 import HomeLayout from '../../components/common/layout/layout';
 import WithdrawPopup from '../../components/mypage/mypage-pop-up/withdraw';
 import { ResignButtonContainer, Section, StyledForm, Title } from '../../styles/mypage/profile';
+import { userIdState } from '../../recoils/atoms/users';
+import { useRecoilValue } from 'recoil';
+import { patchUserInfo, nicknameCheck } from '../../api/fetcher/users';
+import NicknamePopup from '../../components/mypage/mypage-pop-up';
+import { useMutation, useQueryClient } from 'react-query';
+import _, { debounce } from 'lodash';
+
+interface errorMesageType {
+  unknown: string;
+  isError: string;
+  [prop: string]: string;
+}
 
 const Profile = () => {
+  const [nickname, setNickname] = useState<string>('');
+  const [error, setError] = useState<string | undefined>(' ');
```

Comment:
에러타입 기본값 없이는 안되나여 ?! 
```suggestion
  const [nickname, setNickname] = useState<string>('');
  const [error, setError] = useState<string>();
```